pkg/schema/tracker.go

```diff
@@ -329,3 +340,182 @@ func (tr *Tracker) CreateTableIfNotExists(table *filter.Table, ti *model.TableIn
+	if index == nil {
+		log.L().Info(fmt.Sprintf("DownStream schema tracker init: %s.%s", db, table))
+		index = &ToIndexes{
+			schemaName: db,
+			tableName:  table,
+			pks:        make([]string, 0),
+			uks:        make([]string, 0),
+			// uksIsNull:  make([]bool, 0),
+		}
+		// tctx := tcontext.NewContext(ctx, log.With(zap.String("component", "schema-tracker"), zap.String("task", task)))
+		rows, err := tidbConn.QuerySQL(tctx, fmt.Sprintf("SHOW INDEX FROM %s FROM %s", table, db))
+		if err != nil {
+			return nil, err
+		}
+
+		cols, err := rows.Columns()
+		if err != nil {
+			return nil, err
+		}
+		// the column of show statement is too many, so make dynamic values for scan
+		values := make([][]byte, len(cols))
+		scans := make([]interface{}, len(cols))
+		for i := range values {
+			scans[i] = &values[i]
+		}
+
+		for rows.Next() {
+			if err3 := rows.Scan(scans...); err3 != nil {
+				return nil, err3
+			}
+
+			// Key_name -- 2, Column_name -- 4, Null -- 9
+			nonUnique := string(values[1]) // 0 is UK
+			keyName := string(values[2])   // pk is PRIMARY
+			columName := string(values[4])
+			// isNull := string(values[9]) // Null is YES
+
+			if strings.EqualFold(keyName, "PRIMARY") {
+				// handle multiple pk
+				index.pks = append(index.pks, columName)
+				log.L().Info(fmt.Sprintf("DownStream schema tracker %s.%s Find PK %s", db, table, columName))
+			} else if strings.EqualFold(nonUnique, "0") {
+				index.uks = append(index.uks, columName)
+				log.L().Info(fmt.Sprintf("DownStream schema tracker %s.%s Find UK %s ", db, table, columName))
+			}
+		}
+		// nolint:sqlclosecheck
+		if err := rows.Close(); err != nil {
+			return nil, err
+		}
+		if err := rows.Err(); err != nil {
+			return nil, err
+		}
+		tr.toIndexes[db][table] = index
+	}
+
+	// construct model.IndexInfo, PK > not null UK
+	if len(index.pks) != 0 {
+		// handle multiple pk
+		columns := make([]*model.IndexColumn, 0, len(index.pks))
+		for _, pk := range index.pks {
+			if orginColumn := model.FindColumnInfo(originTi.Columns, pk); orginColumn != nil {
+				column := &model.IndexColumn{
+					Name:   model.NewCIStr(pk),
+					Offset: orginColumn.Offset,
+					Length: types.UnspecifiedLength,
+				}
+				columns = append(columns, column)
+			}
+		}
+		if len(columns) != 0 {
+			return &model.IndexInfo{
+				Table:   model.NewCIStr(table),
+				Unique:  true,
+				Primary: true,
+				State:   model.StatePublic,
+				Tp:      model.IndexTypeBtree,
+				Columns: columns,
+			}, nil
+		}
+	}
+	// else if len(index.uks) != 0 {
+	// 	for i := 0; i < len(index.uks); i++ {
+	// 		if !index.uksIsNull[i] {
+	// 			if originColumn := model.FindColumnInfo(originTi.Columns, index.uks[i]); originColumn != nil {
+	// 				return &model.IndexInfo{
+	// 					Table:   model.NewCIStr(table),
+	// 					Unique:  true,
+	// 					Primary: false,
+	// 					State:   model.StatePublic,
+	// 					Tp:      model.IndexTypeBtree,
+	// 					Columns: []*model.IndexColumn{{
```

Comment:
does this kind of comments of code needed? i think it can be removed if is't not needed.
there're many of those in the following.

Reply:
ok，i will remove this kind comments of code.